### PR TITLE
Avoid leaking more popped items

### PR DESCRIPTION
**Pull request by estensen**

### CODE REVIEW

File: core/types/transaction.go [hunk 1]
```diff
@@ -442,13 +442,15 @@ func (t *TransactionsFixedOrder) Peek() Transaction {
...
 	t.Transactions = t.Transactions[1:]
+	t.Transactions[0] = nil // avoid memory leak
 }
 
 // Pop removes the best transaction, *not* replacing it with the next one from
 // the same account. This should be used when a transaction cannot be executed
 // and hence all subsequent ones should be discarded from the same account.
 func (t *TransactionsFixedOrder) Pop() {
 	t.Transactions = t.Transactions[1:]
+	t.Transactions[0] = nil // avoid memory leak
```
Comment (AskAlexSharov) [inline core/types/transaction.go:453]: need tmp variable, this assignment to wrong place

**Review by AskAlexSharov — CHANGES_REQUESTED** [review]

File: core/types/transaction.go (hunk 1 above)
Comment (estensen) [reply core/types/transaction.go]: like this?
https://github.com/ledgerwatch/erigon/pull/6204/commits/31ac2c6e7d06782b34d0a9f0a4de2f4bce5f39dc

Comment (estensen) [reply core/types/transaction.go]: btw, are `Shift()` and `Pop()` supposed to be the same?

Comment (AskAlexSharov) [reply core/types/transaction.go]: @estensen this class code is from Geth, I advise you to see original implementation there.

**Comment by AskAlexSharov** [comment]
@estensen ping

**Comment by AskAlexSharov** [comment]
merging by https://github.com/ledgerwatch/erigon/compare/avoid_leack_2?expand=1

**Pull request closed without merging**